modules/repository-s3/src/internalClusterTest/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryMetricsTests.java

```diff
@@ -0,0 +1,217 @@
+            .build();
+    }
+
+    public void testMetricsWithErrors() throws IOException {
+        final String repository = createRepository(randomRepositoryName());
+
+        final String dataNodeName = internalCluster().getNodeNameThat(DiscoveryNode::canContainData);
+        final var blobStoreRepository = (BlobStoreRepository) internalCluster().getInstance(RepositoriesService.class, dataNodeName)
+            .repository(repository);
+        final BlobStore blobStore = blobStoreRepository.blobStore();
+        final TestTelemetryPlugin plugin = internalCluster().getInstance(PluginsService.class, dataNodeName)
+            .filterPlugins(TestTelemetryPlugin.class)
+            .findFirst()
+            .orElseThrow();
+
+        plugin.resetMeter();
+
+        final OperationPurpose purpose = randomFrom(OperationPurpose.values());
+        final BlobContainer blobContainer = blobStore.blobContainer(BlobPath.EMPTY.add(randomIdentifier()));
+        final String blobName = randomIdentifier();
+
+        // Put a blob
+        final int nPuts = randomIntBetween(1, 3);
+        for (int i = 0; i < nPuts; i++) {
+            final long batch = i + 1;
+            addErrorStatus(INTERNAL_SERVER_ERROR, TOO_MANY_REQUESTS, TOO_MANY_REQUESTS);
+            blobContainer.writeBlob(purpose, blobName, new BytesArray("blob"), false);
+            assertThat(getLongCounterValue(plugin, METRIC_REQUESTS_COUNT, Operation.PUT_OBJECT), equalTo(4L * batch));
+            assertThat(getLongCounterValue(plugin, METRIC_OPERATIONS_COUNT, Operation.PUT_OBJECT), equalTo(batch));
+            assertThat(getLongCounterValue(plugin, METRIC_UNSUCCESSFUL_OPERATIONS_COUNT, Operation.PUT_OBJECT), equalTo(0L));
+            assertThat(getLongCounterValue(plugin, METRIC_EXCEPTIONS_COUNT, Operation.PUT_OBJECT), equalTo(batch));
+            assertThat(getLongCounterValue(plugin, METRIC_THROTTLES_COUNT, Operation.PUT_OBJECT), equalTo(2L * batch));
+            assertThat(getLongHistogramValue(plugin, METRIC_EXCEPTIONS_HISTOGRAM, Operation.PUT_OBJECT), equalTo(batch));
+            assertThat(getLongHistogramValue(plugin, METRIC_THROTTLES_HISTOGRAM, Operation.PUT_OBJECT), equalTo(2L * batch));
+        }
+
+        // Get not found
+        final int nGets = randomIntBetween(1, 3);
+        for (int i = 0; i < nGets; i++) {
+            final long batch = i + 1;
+            addErrorStatus(TOO_MANY_REQUESTS, NOT_FOUND);
+            try {
+                blobContainer.readBlob(purpose, blobName).close();
+            } catch (Exception e) {
+                // intentional failure
+            }
+            assertThat(getLongCounterValue(plugin, METRIC_REQUESTS_COUNT, Operation.GET_OBJECT), equalTo(2L * batch));
+            assertThat(getLongCounterValue(plugin, METRIC_OPERATIONS_COUNT, Operation.GET_OBJECT), equalTo(batch));
+            assertThat(getLongCounterValue(plugin, METRIC_UNSUCCESSFUL_OPERATIONS_COUNT, Operation.GET_OBJECT), equalTo(batch));
```

Comment:
I am surprised that `NOT_FOUND` counts towards unsuccessful operations.
It is not error on s3 side for sure, however we also should not request blobs that do not exists.

Mabye @DaveCTurner have strong opinion about this?

Reply:
404s are _interesting_ because they suggest an ES bug, so definitely worth tracking, but I don't think we should conflate them with other unsuccessful operations indeed.

Reply:
This is a good point. In the same spirit, we could also get 401 or 403 errors for misconfiguration on our side which has happened once. We don't want to count them as unsuccessful operations either if we define "unsuccessful operations" as server side issues (5xx or 429 or similar).

Reply:
Ievgen and I synced and we agreed to leave this as is for now and have a separate discussion on fine tuning the metrics. Also, an option is to leave this as a total count for _all_ unsuccessful operations and add a new one later for just counting the client side unsuccessful operations. They can both have their own use cases.